web/modules/custom/genre_validator/src/Plugin/Validation/Constraint/genreRulesValidator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+
+namespace Drupal\genre_validator\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates the genre_rules Constraint
+ *
+ * @package Drupal\genre_validator\Plugin\Validation\Constraint
+ */
+class genreRulesValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+
+  public function validate($genres, Constraint $constraint) {
+
+    foreach ($genres as $genre) {
+      // Check if the string has any numbers
+      if (preg_match('/[123456789]/', $genre)) {
+        $this->context->addViolation($constraint->noNumbers, ['%value' => $genre->value]);
+      }
+      // Check if the string has any special characters
+      if (preg_match('/[\'^£$%&*()}{@#~?<>,|=_+¬-]/', $genre)) {
+        $this->context->addViolation($constraint->noComma, ['%value' => $genre->value]);
+      }
+      // Check if the first letter is capitalized
+      if (preg_match('/^[A-Z]/', $genre)) {
+        $this->context->addViolation($constraint->notCapitalized, ['%value' => $genre->value]);
+      }
+// Check if every dash has a letter before and after it
+      if (preg_match('/^[A-Za]+(?:-[A-Za]+)?$/D', $genre)) {
```

Comment:
This regexp doesn't work. Check it on https://regex101.com/
Try inputs like `Ab-Cd`, `Ab-Cd Ef`, `Ab Cd`